integ-tests/src/main/java/com/google/testdata/TestDataHelper.java

```diff
@@ -147,4 +147,58 @@ public static void generateConstraints(Connection connection, int constraintsCou
             "Generated %s new constraint(s):\n%s",
             dbTableQueries.size(), Joiner.on("\n").join(dbTableQueries)));
   }
-}
+
+  /**
+   * @param connection DB connection parameter
+   * @param constraintsCount Repetition count
+   */
+  public static void generatePartitioningConstraints(Connection connection, int constraintsCount)
+      throws SQLException {
+    final String partitioningConstraintsData1 = getSql(TESTDATA_SQL_PATH + "columns_data_1.sql");
+    final String partitioningConstraintsData2 = getSql(TESTDATA_SQL_PATH + "partitioning_constraints_data.sql");
+
+    List<String> dbTableQueries = new ArrayList<>();
+    while (constraintsCount > 0) {
+      String dbName = format("test_%s_%s", nanoTime(), constraintsCount);
+      String tableName = format("test_%s_%s", nanoTime(), constraintsCount);
+
+      dbTableQueries.add(format(partitioningConstraintsData1, dbName));
+      dbTableQueries.add(format(partitioningConstraintsData2, dbName, tableName));
+      constraintsCount--;
+    }
+    executeQueries(connection, dbTableQueries);
+
+    LOGGER.info(
+        format(
+            "Generated %s new partitioning constraint(s):\n%s",
+            dbTableQueries.size(), Joiner.on("\n").join(dbTableQueries)));
+  }
+
+  /**
+   * @param connection DB connection parameter
+   * @param statsCount Repetition count
+   */
+  public static void generateStats(Connection connection, int statsCount)
+      throws SQLException {
+    final String statsData1 = getSql(TESTDATA_SQL_PATH + "columns_data_1.sql");
+    final String statsData2 = getSql(TESTDATA_SQL_PATH + "columns_data_2.sql");
+    final String statsData3 = getSql(TESTDATA_SQL_PATH + "stats_data.sql");
+
+    List<String> dbTableQueries = new ArrayList<>();
+    while (statsCount > 0) {
```

Comment:
Any reason you are not using the more common `for (int i = 1; i<=statsCount; i++)`? I know this pattern was once popular in C/C++ due to performance but I doubt it will have measurable impact here.

Reply:
It just requires fewer symbols. Also, I think, with `statsCount--;` it's more transparent and looks less ugly compare with the old school 'for i' cycle.
Maybe it's a matter of preference.